### PR TITLE
[DynamoDB] Update BulkSet and BulkDelete with BatchWriteItem

### PR DESCRIPTION
**Pull request by ksashikumar**
# Description

This PR updates `BulkSet` and `BulkDelete` for DynamoDB with [`BatchWriteItem`](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html)

## Issue reference

This PR will close: https://github.com/dapr/components-contrib/issues/245

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

---
This change is [Reviewable](https://reviewable.io/reviews/dapr/components-contrib/331)

### CODE REVIEW

File: state/aws/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -102,13 +102,32 @@ func (d *StateStore) Set(req *state.SetRequest) error {
...
-		if err != nil {
-			return err
+		writeRequest := &dynamodb.WriteRequest{
+			PutRequest: &dynamodb.PutRequest{
+				Item: map[string]*dynamodb.AttributeValue{
+					"key": {
+						S: aws.String(r.Key),
+					},
+					"value": {
+						S: aws.String(fmt.Sprintf("%v", r.Value)),
```
Comment (artursouza) [inline state/aws/dynamodb/dynamodb.go:115]: I made this comment 9 days ago but forgot to publish. Can you validate this for other types? "S:" means always String.

Comment (ksashikumar) [reply state/aws/dynamodb/dynamodb.go]: @artursouza Thanks for the review 👍 I've updated it, could you please review again?

**Comment by yaron2** [comment]
@ksashikumar the tests are failing

**Comment by artursouza** [comment]
@ksashikumar Unit tests are failing. Can you check it again?

**Comment by ksashikumar** [comment]
@artursouza @yaron2 Sorry for the delay, I've fixed the tests and similar to `BulkSet` we should do the type validation for `state.GetRequest.Value` for `Set` as well. So I've cleaned up the code a bit and updated tests for `Set` also. Could you please review?

**Comment by yaron2** [comment]
Thanks for this @ksashikumar!

**Pull request merged**